controllers/openshift_controller.go

```diff
@@ -300,34 +168,33 @@ WantedBy=multi-user.target
 
 	if kataOC {
 		machinePool = "kata-oc"
-	} else if _, ok := r.kataConfig.Spec.KataConfigPoolSelector.MatchLabels["node-role.kubernetes.io/"+machinePool]; ok {
-		r.Log.Info("in newMCforCR machinePool" + machinePool)
-	} else {
+	} else if _, ok := r.kataConfig.Spec.KataConfigPoolSelector.MatchLabels["node-role.kubernetes.io/"+machinePool]; !ok {
 		r.Log.Error(err, "no valid role for mc found")
 	}
 
 	file := ignTypes.File{}
-	c := ignTypes.FileContents{}
+	c := file.Contents
 
 	dropinConf, err := generateDropinConfig(r.kataConfig.Status.RuntimeClass)
 	if err != nil {
 		return nil, err
 	}
 
-	c.Source = "data:text/plain;charset=utf-8;base64," + dropinConf
+	dropinFile := "data:text/plain;charset=utf-8;base64," + dropinConf
+	c.Source = &dropinFile
+
```

Comment:
We should be able to directly use `file.Contents.Source = &dropinFile`

Reply:
yes, I was going to fix that because I think you already suggested that in another PR. But then I realized that we will delete this code very soon anyway. As soon as the new kata-containers rpm is available that creates the dropin file.